src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/AddOrganisationValidator.cs

```diff
@@ -2,40 +2,238 @@
+        
+        private IRoatpApiClient _apiClient;
+
+        public AddOrganisationValidator(IRoatpApiClient apiClient)
+        {
+            _apiClient = apiClient;
+        }
+
+        public async Task<ValidationResponse> ValidateOrganisationDetails(AddOrganisationViewModel viewModel)
         {
-            var isValid = (providerTypeId >= 1 && providerTypeId <= 3);
+            var validationResponse = new ValidationResponse
+            {
+                Errors = new List<ValidationErrorDetail>()
+            };
+
+            var fieldValidationErrors = IsValidLegalName(viewModel.LegalName);
+            if (fieldValidationErrors.Any())
+            {
+                validationResponse.Errors.AddRange(fieldValidationErrors);
+            }
+
+            fieldValidationErrors = IsValidUKPRN(viewModel.UKPRN);
+            if (fieldValidationErrors.Any())
+            {
+                validationResponse.Errors.AddRange(fieldValidationErrors);
+            }
+            else
+            {
+                fieldValidationErrors = IsDuplicateUKPRN(viewModel.OrganisationId, viewModel.UKPRN);
+                if (fieldValidationErrors.Any())
+                {
+                    validationResponse.Errors.AddRange(fieldValidationErrors);
+                }
+            }
+
+            fieldValidationErrors = IsValidCompanyNumber(viewModel.CompanyNumber);
+            if (fieldValidationErrors.Any())
+            {         
+                validationResponse.Errors.AddRange(fieldValidationErrors);    
+            }
+            else
+            {
+                fieldValidationErrors = IsDuplicateCompanyNumber(viewModel.OrganisationId, viewModel.CompanyNumber);
+                if (fieldValidationErrors.Any())
+                {
+                    validationResponse.Errors.AddRange(fieldValidationErrors);
+                }
+            }
 
-            if (!isValid)
+            fieldValidationErrors = IsValidCharityNumber(viewModel.CharityNumber);
+            if (fieldValidationErrors.Any())
+            {
+                validationResponse.Errors.AddRange(fieldValidationErrors);
+            }
+            else
             {
-                return new List<string>
+                fieldValidationErrors = IsDuplicateCharityNumber(viewModel.OrganisationId, viewModel.CharityNumber);
+                if (fieldValidationErrors.Any())
                 {
-                    AddOrganisationValidationErrorMessages.InvalidProviderTypeId
-                };
-            } 
+                    validationResponse.Errors.AddRange(fieldValidationErrors);
+                }
+            }
+
+            fieldValidationErrors = IsValidTradingName(viewModel.TradingName);
+            if (fieldValidationErrors.Any())
+            {
+                validationResponse.Errors.AddRange(fieldValidationErrors);
+            }
+
+            return validationResponse;
+        }
+        
```

Comment:
See previous comment about isolating these rules into their own service.  Let's create a tech debt story about it soon.